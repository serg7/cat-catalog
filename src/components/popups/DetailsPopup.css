/* .popup {
  display: none;
  position: absolute;
  z-index: 1;
  top: 20%;
  width: 75%;
  overflow: auto;
  border: 1px solid #aaa;
  background-color: rgb(231, 231, 231);
  -webkit-animation-name: fadeIn;
  -webkit-animation-duration: 0.4s;
  animation-name: fadeIn;
  animation-duration: 0.4s;
}

.popup .popup-content {
  justify-content: center;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  align-items: stretch;
  background-color: #fefefe;
  width: 100%;
  -webkit-animation-name: slideIn;
  -webkit-animation-duration: 0.4s;
  animation-name: slideIn;
  animation-duration: 0.4s;
} */

.popup .show {
  visibility: visible;
  -webkit-animation: fadeIn 1s;
  animation: fadeIn 1s;
}

.close {
  color: white;
  font-size: 28px;
  font-weight: bold;
  position: relative;
  top: -10px;
  right: -280px;
}

.close:hover,
.close:focus {
  color: rgb(53, 50, 50);
  text-decoration: none;
  cursor: pointer;
}

.levels {
    border: 1px solid #bbb;
    margin: 10px;
}

.feature-container {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 99%;
  padding: 3px;
  background-color: #f1f1f1;
}

.feature {
  width: 20%;
  margin-right: 20px;
  margin-left: 20px;
}

.level-5 {
  width: 60%;
  height: 18px;
  background-color: #04aa6d;
}
.level-4 {
  width: 30%;
  height: 18px;
  background-color: #2196f3;
}
.level-3 {
  width: 15%;
  height: 18px;
  background-color: #00bcd4;
}
.level-2 {
  width: 10%;
  height: 18px;
  background-color: #ff9800;
}
.level-1 {
  width: 4%;
  height: 18px;
  background-color: #f44336;
}

.header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 10px;
    border: 1px solid #aaa;
}
.short-info {
    width: 30%;
}
.name {
    font-weight: bold;
}
.image { 
    height: 200px;
}
.image:hover {
    cursor: pointer;
}


@keyframes slideIn {
  from {
    bottom: -300px;
    opacity: 0;
  }
  to {
    bottom: 0;
    opacity: 1;
  }
}

@-webkit-keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@-webkit-keyframes slideIn {
    from {
      bottom: -300px;
      opacity: 0;
    }
    to {
      bottom: 0;
      opacity: 1;
    }
  }
