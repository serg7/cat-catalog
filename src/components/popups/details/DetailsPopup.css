.levels {
    border: 1px solid #bbb;
    margin: 10px;
}

.feature-container {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 99%;
  padding: 3px;
  background-color: #f1f1f1;
}

.feature {
  width: 20%;
  margin-right: 20px;
  margin-left: 20px;
}

.level-5 {
  width: 60%;
  height: 18px;
  background-color: #04aa6d;
}
.level-4 {
  width: 30%;
  height: 18px;
  background-color: #2196f3;
}
.level-3 {
  width: 15%;
  height: 18px;
  background-color: #00bcd4;
}
.level-2 {
  width: 10%;
  height: 18px;
  background-color: #ff9800;
}
.level-1 {
  width: 4%;
  height: 18px;
  background-color: #f44336;
}

.header {
    display: flex;
    align-items: center;
    justify-content: space-around;
    padding: 10px;
    border: 1px solid #aaa;
}
.short-info {
    width: 30%;
}
.name {
    font-weight: bold;
}
.image { 
    height: 200px;
}
.image:hover {
    cursor: pointer;
}


@keyframes slideIn {
  from {
    bottom: -300px;
    opacity: 0;
  }
  to {
    bottom: 0;
    opacity: 1;
  }
}

@-webkit-keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@-webkit-keyframes slideIn {
    from {
      bottom: -300px;
      opacity: 0;
    }
    to {
      bottom: 0;
      opacity: 1;
    }
  }
