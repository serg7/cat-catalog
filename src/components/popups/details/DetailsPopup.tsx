import React from 'react';
import classNames from 'classnames';
import './DetailsPopup.css';
import '../CommonPopup.css';
import featuresWithLevel from './Features';
import Cat from './Cat';

export const DetailsPopup = (props: { show: boolean; setIsShown: (isShown: boolean) => void;  cat: Cat }): JSX.Element => {
  const { cat, show, setIsShown } = props;

  const renderFeatureLevels = (cat: any): JSX.Element[] =>
    featuresWithLevel.map((feature: string, index: number) => (
      <div className="feature-container" key={index}>
        <div className="feature">{feature}</div>
        <div className={`level-${cat[feature]}`}></div>
      </div>
    ));

  return (
    <div className={classNames('popup', { show })} id="modal" data-testid="details-popup">
      <span className="close" onClick={() => setIsShown(false)}>
        &times;
      </span>
      <div className="popup-content">
        <div className="header">
          <div className="short-info">
            <div className="name">{cat.name}</div>
            <div className="origin">Origin: {cat.origin}</div>
          </div>
          <img className="image" src={cat?.image?.url} />
        </div>
        <div className="levels">{renderFeatureLevels(cat)}</div>
      </div>
    </div>
  );
};

export default DetailsPopup;
