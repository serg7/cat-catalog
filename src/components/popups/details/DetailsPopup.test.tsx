import React from 'react';
import { render, screen } from '@testing-library/react';
import DetailsPopup from './DetailsPopup';
import Cat from '../details/Cat';

test('should render Detail popup', () => {
  render(<DetailsPopup show={true} cat={{} as Cat} />);
  const popup = screen.getByTestId('details-popup');

  expect(popup).toMatchSnapshot();
});