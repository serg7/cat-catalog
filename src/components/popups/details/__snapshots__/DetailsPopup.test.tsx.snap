// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Detail popup 1`] = `
<div
  class="popup show"
  data-testid="details-popup"
  id="modal"
>
  <span
    class="close"
  >
    Ã—
  </span>
  <div
    class="popup-content"
  >
    <div
      class="header"
    >
      <div
        class="short-info"
      >
        <div
          class="name"
        />
        <div
          class="origin"
        >
          Origin: 
        </div>
      </div>
      <img
        class="image"
      />
    </div>
    <div
      class="levels"
    >
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          adaptability
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          affection_level
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          child_friendly
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          dog_friendly
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          energy_level
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          health_issues
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          intelligence
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          shedding_level
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          social_needs
        </div>
        <div
          class="level-undefined"
        />
      </div>
      <div
        class="feature-container"
      >
        <div
          class="feature"
        >
          stranger_friendly
        </div>
        <div
          class="level-undefined"
        />
      </div>
    </div>
  </div>
</div>
`;
